pkg/cvo/sync_worker.go

```diff
@@ -58,6 +58,12 @@ type PayloadRetriever interface {
 	RetrievePayload(ctx context.Context, desired configv1.Update) (PayloadInfo, error)
 }
 
+// ClusterProfileRetriever abstracts how the cluster profile is retrieved.
+type ClusterProfileRetriever interface {
+	// RetrieveClusterProfile attempts to retrieve the profile
+	RetrieveClusterProfile(ctx context.Context) (string, error)
```

Comment:
I'm not a huge fan of `interface` that wrap a single function.  And I think the risk of needing to grow a new function in the future is small.  Can we make this:

```go
// RetrieveClusterProfile retrieves the cluster profile.
type RetrieveClusterProfile func(ctx context.Context) (string, error)
```

That would allow us to drop `fakeClusterProfileRetriever` and just inline `func (ctx context.Context) {return "default", nil}` where we needed a dummy.

Reply:
done :+1: